[test] Fix attr_implements_serial.swift harder

jrose-apple opened the pull request:
Building dylibs that can be moved around later is tricky; on Apple platforms, you have to make sure the "install name" isn't an absolute path. We already have a substitution to handle this, so use it.

jrose-apple commented:
@swift-ci Please test

jrose-apple commented:
@swift-ci Please test Linux

slavapestov commented:
@graydon Executable tests should be in test/Interpreter/ -- can you redo this please?

jrose-apple committed:
[test] Fix attr_implements_serial.swift harder

Building dylibs that can be moved around later is tricky; on Apple
platforms, you have to make sure the "install name" isn't an absolute
path. We already have a substitution to handle this, so use it.

jrose-apple commented:
@swift-ci Please smoke test
